app/src/router.js

```diff
@@ -7,35 +7,33 @@ import { setNetwork } from "./scripts/setNetwork"
 Vue.use(router)
 
 export const routeGuard = (store) => async (to, from, next) => {
-  // Set any open modal to false
+  // set any open modal to false
   store.state.session.currrentModalOpen = false
 
-  // notifications is landing page when user is signed in with email
-  if (to.fullPath.includes("portfolio") && store.state.account.userSignedIn) {
+  // notifications is landing page if user is signed in with email
+  if (store.state.account.userSignedIn) {
     next(`/notifications`)
   }
-  // Redirect if fullPath begins with a hash (fallback for old pre history mode urls)
+
+  // fallback for old history mode url redirect if path includes a hash
   if (to.fullPath.includes("#")) {
     const path = to.fullPath.substr(to.fullPath.indexOf("#") + 1)
     next(path)
-    return
   }
-  if (to.name === `notifications`) {
-    if (store.state.account.userSignedIn) {
-      next()
-      return
-    } else if (from.name === `paywall`) {
-      return
-    } else {
-      next(`/paywall`)
-    }
+
+  // redirect from notifications to paywall is user is not signed in
+  if (to.name === `notifications` && !store.state.account.userSignedIn) {
+    next(`/paywall`)
```

Comment:
Well, it took me a while to fix this route. Maybe I should have left some comment about this, as it is not obvious. But if you are already in `/paywall` and you click on the notifications icon, then you are taken to notifications no matter what. That's why I had to add that `else if`:wink: It wasn't by whim